plugins/categoryaslink/class.categoryaslink.plugin.php

```diff
@@ -118,8 +118,19 @@ function categoryUrl($category, $page = '', $withDomain = true) {
 
         // If there is a URL that links to a discussion it overrides the category or even a link to an alias category.
         if (val('RedirectUrl', $category)) {
-            // SafeURL because you may be linking to another web property, another forum or knowledgebase.
-            return safeURL(val('RedirectUrl', $category));
+            // If SubcommunitiesPlugin is enabled we have to remove "community root directory of the request" in URL.
+            if (class_exists('SubcommunitiesPlugin')) {
+                $requestRoot = Gdn::request()->getRoot();
+                // Remove $requestRoot in $destination because it will appear twice because of url() or safeURL().
+                $destination = safeURL($category['RedirectUrl']);
+                $redirectUrlDestination = $requestRoot && strpos($destination, $requestRoot) === 0 ?
+                    substr($destination, strlen($requestRoot)) : $destination;
+            } else {
+                // SafeURL because you may be linking to another web property, another forum or knowledgebase.
+                $redirectUrlDestination = safeURL(val('RedirectUrl', $category));
+            }
+
+            return $redirectUrlDestination;
```

Comment:
This is not really a sufficient solution.

Subcommunities is not the only thing we have that can modify the base community URL. (hub/node, reverseproxy plugin).

Additionally this is not the only plugin using `safeURL()`. Maybe `safeURL` is what should be updated here.

No core plugin should **_ever_** be doing `if (SubcommunityPlugin)`. It's a big hack, and isn't particularly robust.

Slack me if you want help finding a better solution.

Reply:
@charrondev I used `if (class_exists('SubcommunitiesPlugin')) {` because of existing  code with the condition.  `safeURL`  is used twice because of the redirectionTo in`categoriesController_beforeCategoriesRender_handler`. I was thinking of historical reasons  it was  like that. In fact we can return `$category['RedirectUrl']` directly

Reply:
That works. Do you know if that condition was in the product or in the services repos?

This type of plugin checking is generally verboten on the product side, as it leads to very fragile code. You can sometimes get away with it in a singular customer repo because the customer has a much narrower set of site configuration possbilities.

Reply:
You mean this kind of code? 
```php      
if (class_exists('SubcommunitiesPlugin')) {
            // Subcommunities version.
            $path = '/categories/'.rawurlencode(val('UrlCode', $category));
            $categoryURL = Gdn::getContainer()
                ->get(SubcommunitiesPlugin::class)
                ->subcommunityURL($category['categoryID'] ?? 0, $path, $withDomain, $page, SubcommunitiesPlugin::URL_TYPE_CATEGORY);
        } elseif (class_exists('SEOLinksPlugin')) {
```

Reply:
@charrondev `class_exists('SubcommunitiesPlugin')` was already in the code. **Categories as Links** is an old plugin.

This PR is related to [this issue](https://github.com/vanilla/support/issues/1726) which was escalated to Services because it started presenting issues after Asus started using a custom [redirector addon](https://github.com/vanillaforums/asus/tree/master/plugins/zentalkredirector) but since it seems it's core-related maybe it shouldn't be up to Services to fix. Also seems more complex than a tweak in `categoryaslink.plugin.php` as you mentioned so maybe we should close this PR and start over. ICB can take the issue back.

Reply:
The thing is `CategoryAsLink` isn't that old of a plugin. It's only from the last year and a half.

In any case that sounds like a good idea.